examples/cpp/hello_world/README.md

```diff
@@ -0,0 +1,184 @@
+
+In this case, the *hello world* example describes the simplest deployment of a Fast DDS publisher and subscriber.
+
+## Example in deep
+
+Each entity of the example (publisher and subscriber) creates different nested DDS entities: domain participant, publisher & dataWriter, and subscriber & dataReader, respectively.
+In both cases, the three DDS entities (domain participant, publisher/subscriber and dataWriter/dataReader) load their default configuration from the environment.
+If the environment does not specify the expected configuration, they take the default configuration per entity.
+For further information regarding the configuration environment, please refer to the *[XML profile playground](#xml-profile-playground)* subsection in the *[advanced configuration](#advanced-configuration)* section.
+
+The *hello world* example, together with the remain examples, would include a listening callback on the subscriber side. The subscriber will manage the new available data in the same thread as the main subscriber application.
+For simplicity, the ``HelloWorldSubscriber`` implements it own callback.
+
+Furthermore, this particular example includes a wait-set implementation. In contrast to implementing a listening callback, the wait-set is a mechanism where a dedicated thread waits until a status condition occurs. In that moment, that status condition triggering event would be evaluated to determine witch actions should be taken against it.
+
+For this example, both listening callback and wait-set implementation would run the same code and generate the same output for both triggering events: subscription matching and new data available.
+
+## Run the example
+
+To launch this test open two different consoles. One of them will run the publisher side of the example, and the other would run the subscriber side.
+
+### Hello world publisher
+
+#### Ubuntu ( / MacOS )
+
+```shell
+user@machine:example_path$ ./hello_world publisher
+Publisher running. Please press Ctrl+C to stop the Publisher at any time.
+```
+
+#### Windows
+
+```powershell
+example_path> hello_world.exe publisher
+Publisher running. Please press Ctrl+C to stop the Publisher at any time.
+```
+
+### Hello world subscriber
+
+#### Ubuntu ( / MacOS )
+
+```shell
+user@machine:example_path$ ./hello_world subscriber
+Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
+```
+
+#### Windows
+
+```powershell
+example_path> hello_world.exe subscriber
+Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
+```
+
+## Expected output
+
+It does not matter which entity is launched first, because the publisher has a condition to wait until the first match to start sending hello world samples.
+The expected output on both cases is a first displayed message acknowledging they have matched, followed by the amount of samples sent or received until user press Ctrl+C.
```

Comment:
```suggestion
Regardless of which application is run first, since the published will not start sending that until a subscriber is discovered, the expected output both for publishers and subscriber is a first displayed message acknowledging the match, followed by the amount of samples sent or received until Ctrl+C is pressed
```